edgelet/iotedged/src/workload_manager.rs

```diff
@@ -246,6 +246,9 @@ where
     fn remove(&mut self, module_id: &str) -> Result<(), Error> {
         info!("Removing listener for module {}", module_id);
 
+        //Try to stop the listener, in case it was not stopped before.
+        self.stop(module_id).ok();
```

Comment:
What's this `.ok()` for? If you want to ignore the `Err` result, then do `let _ = self.stop(module_id);` instead, with a comment explaining why you want to ignore the `Err` result.

But more importantly, it would be better to a) change `stop` so that calling it multiple times is not an error, ie it returns `Ok(())` if `shutdown_sender` is `None`, and b) holding on to the error from `stop` and returning that from `remove` after the cleanup is done so that `remove` also fails.

Also, I see now that the `Result`s returned from these functions are entirely ignored because `fn server` just does `.unwrap_or(())` on them. That should be changed to log in the case of an error.

Reply:
Good point, I will modified 1.2 and edgelet refactor with this.